src/index.js

```diff
@@ -150,8 +155,101 @@ const renderLevel = (root, levelNumber) => {
         .join("")}
     </tr>
   `;
+
+  renderSettings = () => {  
+    const settingsHTML = `
+    <div id="modal">
+      <div class="modal-content">
+        <span class="close material-symbols-outlined">cancel</span>
+        <h1>Setting</h1>
+      </div>
+    </div>
+    `;
+    document.querySelector("#overlay-screen").innerHTML = settingsHTML;
+
+    document.querySelector(".close").addEventListener("click", () => {
+      document.querySelector("#modal").style.display = "none";
+    });
+  };
+
+  renderHelp = () => {  
+    const helpHTML = `
+    <div id="modal">
+      <div class="modal-content">
+        <span class="close material-symbols-outlined">cancel</span>
+        <h1>How To Play</h1>
+        <p>Objective: Push the boxes to all of the goals to complete the level in the least amount of moves.</p>
+        <p>Use the mouse or keyboard to move around the board.</p>
+        <p>Keyboard Shortcuts:</br>
+          direction: arrow keys;  undo: z;  reset: r
+        </p>
+      </div>
+    </div>
+    `;
+    document.querySelector("#overlay-screen").innerHTML = helpHTML;
+
+    document.querySelector(".close").addEventListener("click", () => {
+      document.querySelector("#modal").style.display = "none";
+    });
+  };
+
+  renderLevelComplete = () => {
+    const completeHTML = `
+    <div id="modal">
+      <div class="modal-content"> 
+        <h1 class="text-glow">Level ${soko.levelNumber()+1}</h>
+        <h1 class="text-glow">Complete!</h1>
+        <table>
+          <tr>
+            <td>Current moves:</td>
+            <td id="current-move" class="score">${soko.movesSize()}</td>
+          </tr>
+          <tr>
+            <td>Best moves:</td>
+            <td id="best-move" class="score">${localStorage.getItem(soko.levelNumber()+1)}
+            </td>
+          </tr>
+        </table>
+        <div id="controls">
+          <button title="Random" id="random-level"><span class="material-symbols-outlined">shuffle</span></button>
+          <button title="Home" id="change-level"><span class="material-symbols-outlined">home</span></button>
+          <button title="Next" id="next"><span class="material-symbols-outlined">fast_forward</span></button>
+        </div>
+    </div>
+    `;
+    document.querySelector("#overlay-screen").innerHTML = completeHTML;
+
+    /* Listens to home selector */
+    document.querySelector("#change-level").addEventListener("click", () => {
+      location.hash = "";
+      renderMenu(root);
+    })
+    ;
+
+    /* Listens to next level selector */
+    document.querySelector("#next").addEventListener("click", () => {
+      if (levelNumber < (soko.levelsSize() - 1)) {
+        location.hash = (levelNumber + 2).toString();
```

Comment:
The `+ 2` is pretty confusing here. I suggest we follow these rules:

- whenever we store the current level in a variable or use it internally for any code logic, it's 0 indexed.
- whenever we read the level number from the UI (e.g. `location.hash`), we substract 1 immediately before storing it in a var.
- whenever we need to show the current level number (or any level number on the UI), we add 1 just before emitting it to the user.

This way, we're always 100% 0-indexed in terms of our code logic. So `renderLevel` and all JS logic expects 0-indexing, as does the underlying C++ code.